connectors/src/api/get_connector_permissions.ts

```diff
@@ -55,11 +55,12 @@ const _getConnectorPermissions = async (
   const connectorPermissionRetriever =
     RETRIEVE_CONNECTOR_PERMISSIONS_BY_TYPE[connector.type];
 
-  const pRes = await connectorPermissionRetriever(
-    connector.id,
+  const pRes = await connectorPermissionRetriever({
+    connectorId: connector.id,
     parentInternalId,
-    filterPermission
-  );
+    filterPermission,
+    retrieveAncestors: req.query.retrieveAncestors === "true",
```

Comment:
Should it be called retrieveParents?

Reply:
What's the rationale for introducing the `ancestor` name?

Reply:
`Parents` is less clear than ancestor IMHO, as it implies only 1 level of depth. Ancestor make it obvious that it's multi-level;

Reply:
Definitely loosely held, I can switch to parents if needed

Reply:
I personally don't read ancestors as necessarilly deeper than parents in both case it's not obvious of course that we get the full path in general (if having multiple parents is possible). But it's kid of assumed by the rest of the API (internalParentId being a string not an array) that there is only one direct parent. So parents kind of clearly states that we're going up to the root IMHO

I think adding a new name is more confusing than using parents here.